### ISSUE TITLE
Thumbnail cut from top and buttom in video page

### ISSUE THREAD

**Issue by rezad1393** [iv-org/invidious#2237]
**Describe the bug**
the thumbnail in the video page has top and bottom of it cut compared to video thumbnail in search and also in youtube page.

link https://www.youtube.com/watch?v=SDqGpKpTnUg

**Screenshots**
![Screenshot 2021-07-03 at 03-59-35 CBDC Will Control The Way You Spend Money](https://user-images.githubusercontent.com/7523164/124336408-2e754b80-dbb3-11eb-89d4-7bee6da5afc9.png)

![Screenshot 2021-07-03 at 03-59-40 CBDC Will Control The Way You Spend Money](https://user-images.githubusercontent.com/7523164/124336399-2ae1c480-dbb3-11eb-9281-047d6bf60182.png)

**Comment by SamantazFox** [comment]
This feature is working as intended :)

In order to be compatible with many browsers and to run without Javascript, Invidious relies on native browser features. For instance, to display the video thumbnail we use [the `poster` attribute of the `<video>` HTML element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/video#attr-poster)).

Youtube, on the other hand, relies on lots of javascript code and HTML workarounds, without caring about compatibility nor standards (they want you to install Chrome, after all) in order to provide their own "shiny" player.

**Comment by rezad1393** [comment]
so firefox implementation or html standard?
I tested in brave and it is the same as firefox 
tested in edge on windows too.
on edge it is cropped too but more than firefox and brave.

it this a bug( bad number used for size of picture) of html standard or any browser uses its own definition?
can I ask firefox to change this or it is futile??

**Comment by SamantazFox** [comment]
Wait, this only happens when Javascript is enabled!

without JS, there is the behavior that I explained (standard HTML features):
![image](https://user-images.githubusercontent.com/52980486/124386151-4fd05780-dcd9-11eb-9ab6-7b4da2b878fc.png)

And with JS enabled, here is your bug:
![Uploading image.png…]()

**Comment by SamantazFox** [comment]
after further investigation: VideoJS adds its own "poster" image above the player, similar to what youtube does. And as the player doesn't have a 16:9 ratio (black bars on the javascript-less version) the miniature on top is cropped to fit in this space.

**Comment by rezad1393** [comment]
so the firefox/html one (without javascript) would show the correct (complete) poster like youtube and with javascript then invidious javascript player cuts it?
that would be a invidious bug. right?

**Comment by SamantazFox** [comment]
> so the firefox/html one (without javascript) would show the correct (complete) poster like youtube and with javascript then invidious javascript player cuts it?
> that would be a invidious bug. right?

I'm not sure about that. The native player dimensions being forced (with black bars on the side) is the reason why the javacript player image is cropped. I don't know what were the original intentions in this design. Maybe @syeopite knows more.

There is also that issue that may be related: #1880

**Comment by rezad1393** [comment]
I dont think it is the native fault if the invidious javascript player shows the thumbnail wrong.
isnt the javascript one written by invidious team? so they can make the poster whatever they want.

btw I am not familiar with native player capabilities. does it support dash or does dash require JS?

**Comment by SamantazFox** [comment]
> isnt the javascript one written by invidious team? so they can make the poster whatever they want.

The javascript player is called [VideoJS](https://github.com/videojs/video.js), and it's developped by another team, not related to the invidious project.

> btw I am not familiar with native player capabilities. does it support dash or does dash require JS?

As far as I know, DASH requires a javascript player.

**Comment by syeopite** [comment]
> > so the firefox/html one (without javascript) would show the correct (complete) poster like youtube and with javascript then invidious javascript player cuts it?
> > that would be a invidious bug. right?
> 
> I'm not sure about that. The native player dimensions being forced (with black bars on the side) is the reason why the javacript player image is cropped. I don't know what were the original intentions in this design. Maybe @syeopite knows more.
> 
> There is also that issue that may be related: #1880

Sorry for the late reply. But afaik the only reason why the player is like that is due to the upstream bug — which is outsourced to the end users — regarding container size.

Invidious has a few CSS that attempts to resolve the issue but they really don't do much.

**Comment by rezad1393** [comment]
> > isnt the javascript one written by invidious team? so they can make the poster whatever they want.
> 
> The javascript player is called [VideoJS](https://github.com/videojs/video.js), and it's developped by another team, not related to the invidious project.

cant a issue be opened in that project page?

**Comment by syeopite** [comment]
Unfortunately not. Although the bug is an upstream one, they're outsourcing the actual "fixing" to the end user; the progress of which is tracked here: #1880

**Comment by rezad1393** [comment]
does that mean this project need to apply a hack over the videojs source for it to work?
it is too complicated?

**Comment by syeopite** [comment]
Nope. A CSS or configuration fix should theoretically resolve it; this is currently one of my goals for the eventual UI rewrite I'm doing.

**Comment by rezad1393** [comment]
> Nope. A CSS or configuration fix should theoretically resolve it. This is currently one of my goals for the eventual UI rewrite I'm doing.

thank you.
any roadmap for this issue?
or is it to dependent on other issues that it is not worth fixing it right now and we should wait for other issues related to UI to be decided first?
